components/match2/wikis/crossfire/match_group_input_custom.lua

```diff
@@ -0,0 +1,490 @@
+	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_publishertier'))
+
+	match = MatchGroupInput.getCommonTournamentVars(match)
+
+	match.game = _GAME[match.game or '']
+
+	return match
+end
+
+function matchFunctions.getVodStuff(match)
+	match.stream = Streams.processStreams(match)
+	match.vod = Logic.emptyOr(match.vod, Variables.varDefault('vod'))
+
+	match.lrthread = Logic.emptyOr(match.lrthread, Variables.varDefault('lrthread'))
+
+	match.links = {}
+	local links = match.links
+	if match.preview then links.preview = match.preview end
+	if match.esl then links.esl = 'https://play.eslgaming.com/match/' .. match.esl end
+	if match.faceit then links.faceit = 'https://www.faceit.com/en/halo_infinite/room/' .. match.faceit end
+	if match.stats then links.stats = match.stats end
+
+	return match
+end
+
+function matchFunctions.getExtraData(match)
+	match.extradata = {
+		mvp = matchFunctions.getMVP(match),
+		isconverted = 0
+	}
+	return match
+end
+
+-- Parse MVP input
+function matchFunctions.getMVP(match)
+	if not match.mvp then return {} end
+	local mvppoints = match.mvppoints or 1
+
+	-- Split the input
+	local players = mw.text.split(match.mvp, ',')
+
+	-- Trim the input
+	for index,player in pairs(players) do
+		players[index] = mw.text.trim(player)
+	end
+
+	return {players=players, points=mvppoints}
+end
+
+function matchFunctions.getOpponents(match)
+	-- read opponents and ignore empty ones
+	local opponents = {}
+	local isScoreSet = false
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		-- read opponent
+		local opponent = match['opponent' .. opponentIndex]
+		if not Logic.isEmpty(opponent) then
+			CustomMatchGroupInput.processOpponent(opponent, match.date)
+
+			-- Retrieve icon for team
+			if opponent.type == Opponent.team then
+				opponent.icon, opponent.icondark = opponentFunctions.getIcon(opponent.template)
+			end
+
+			-- apply status
+			if TypeUtil.isNumeric(opponent.score) then
+				opponent.status = 'S'
+				isScoreSet = true
+			elseif Table.includes(ALLOWED_STATUSES, opponent.score) then
+				opponent.status = opponent.score
+				opponent.score = -1
+			end
+			opponents[opponentIndex] = opponent
+
+			-- get players from vars for teams
+			if opponent.type == 'team' and not Logic.isEmpty(opponent.name) then
+				match = matchFunctions.getPlayers(match, opponentIndex, opponent.name)
+			end
+		end
+	end
+
+	-- see if match should actually be finished if bestof limit was reached
+	if isScoreSet and not Logic.readBool(match.finished) then
+		local firstTo = math.ceil(match.bestof/2)
+		for _, item in pairs(opponents) do
+			if tonumber(item.score or 0) >= firstTo then
+				match.finished = true
+				break
+			end
+		end
+	end
+
+	-- see if match should actually be finished if score is set
+	if isScoreSet and not Logic.readBool(match.finished) and match.hasDate then
+		local currentUnixTime = os.time(os.date('!*t'))
```

Comment:
make const var

Reply:
How can I solve this one? Currently I had this revision (not on git yet) https://liquipedia.net/crossfire/index.php?title=Module%3AMatchGroup%2FInput%2FCustom%2Fdev&type=revision&diff=21122&oldid=21008 it returns 'match' as a nil value. Do I need an extra const var for **match** ?

since I assume I have to moved it out of the ifs and placed top of module instead?